controllers/serviceaccount_controller.go

```diff
@@ -835,11 +835,9 @@ func renewSecret(
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *GrafanaServiceAccountReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	// TODO: Consider watching token Secrets for reactive reconciles.
-	// It'll requeue on Secret create/update/delete, reducing reliance on ResyncPeriod.
-	// Example: Owns(&corev1.Secret{}, builder.WithPredicates(...))
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1beta1.GrafanaServiceAccount{}).
+		Owns(&corev1.Secret{}).
 		WithEventFilter(predicate.Or(
 			ignoreStatusUpdates(),
 			predicate.AnnotationChangedPredicate{},
```

Comment:
The `ignoreStatusUpdates` is effectively a check to see if `.metadata.generation` has changed.
But Secrets do not have a generation field that is updated on changes.
The controller wide event filter prevents secrets from triggering a reconcile, we would need to move the filter predicates into the `For(...)` args similar to the `Grafanas` Controller.
https://github.com/grafana/grafana-operator/blob/2f3f3547ebf3a7bc8a77e40bc7a8433b10245bcc/controllers/grafana_controller.go#L301-L306

Additionally, we no longer need the `predicate.AnnotationChangedPredicate{}` as the E2E tests no longer rely on it!

Reply:
Hey @Baarsgaard 

Thanks for explaining. I was not aware of that. I have updated em